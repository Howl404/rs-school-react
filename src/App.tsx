import Router from 'src/router/Router';

import 'src/App.scss';

export default function App() {
  return <Router />;
}
