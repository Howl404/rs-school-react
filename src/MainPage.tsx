import React, { Component } from 'react';

class MainPage extends Component {
  state = {
    searchTerm: '',
  };

  handleSearch = (searchTerm: string) => {
    this.setState({ searchTerm });
  };

  render() {
    return <></>;
  }
}

export default MainPage;
