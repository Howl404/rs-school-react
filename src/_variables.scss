$black: black;
$white: white;
$black-shadow: hsl(206 22% 7% / 35%) 0 0 38px -10px;
$grey: #8c92ac44;
