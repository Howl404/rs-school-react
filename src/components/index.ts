export { PasswordStrength } from './PasswordStrength/PasswordStrength';
export { AutoComplete } from './AutoComplete/AutoComplete';
