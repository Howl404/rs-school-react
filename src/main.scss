@import 'src/fonts/fonts.scss';

body {
  margin: 0;
  font-family: $font-family-primary;
  min-height: 100vh;
}

input,
p {
  font-family: $font-family-secondary;
}
