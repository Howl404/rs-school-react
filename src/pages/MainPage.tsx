import { useState } from 'react';
import Search from 'src/components/search/Search';
import Results from 'src/components/results/Results';
import styles from 'src/pages/MainPage.module.scss';

function MainPage() {
  const [searchTerm, setSearchTerm] = useState<string | null>(null);
  const [error, setError] = useState(false);

  if (error) {
    throw new Error('This is a test error');
  }

  return (
    <>
      <button
        className={styles.button}
        onClick={() => {
          setError(true);
        }}
      >
        Throw Error
      </button>
      <Search setSearchTerm={setSearchTerm} searchTerm={searchTerm} />
      <Results searchTerm={searchTerm} />
    </>
  );
}

export default MainPage;
