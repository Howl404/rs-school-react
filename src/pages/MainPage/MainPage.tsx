import { useState } from 'react';

import { SearchTermContextProvider } from 'src/contexts/SearchTermContext';
import { ProductsContextProvider } from 'src/contexts/ProductsContext';

import Search from 'src/components/search/Search';
import Results from 'src/components/results/Results';

import styles from 'src/pages/MainPage/MainPage.module.scss';
import { DetailedProductContextProvider } from 'src/contexts/DetailedProductContext';

export default function MainPage() {
  const [error, setError] = useState(false);

  const throwError = () => {
    setError(true);
  };

  if (error) {
    throw new Error('This is a test error');
  }

  return (
    <>
      <button
        className={styles.errorButton}
        onClick={throwError}
        data-testid="error-button"
      >
        Throw Error
      </button>
      <SearchTermContextProvider>
        <Search />
        <ProductsContextProvider>
          <DetailedProductContextProvider>
            <Results />
          </DetailedProductContextProvider>
        </ProductsContextProvider>
      </SearchTermContextProvider>
    </>
  );
}
