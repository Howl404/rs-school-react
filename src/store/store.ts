import { configureStore } from '@reduxjs/toolkit';
import { createWrapper } from 'next-redux-wrapper';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

import { apiService } from 'store/api/api';

export const store = () =>
  configureStore({
    reducer: {
      [apiService.reducerPath]: apiService.reducer,
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(apiService.middleware),
  });

export type AppStore = ReturnType<typeof store>;

export type RootState = ReturnType<AppStore['getState']>;

export type AppDispatch = AppStore['dispatch'];

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export const wrapper = createWrapper<AppStore>(store);
