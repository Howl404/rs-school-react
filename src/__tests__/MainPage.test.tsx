import { render, screen } from '@testing-library/react';
import { mockProducts } from 'mock/handler';
import { Provider } from 'react-redux';
import { store } from 'src/store';
import { expect, it } from 'vitest';

import MainPage from '../pages/index';

it('Ensure that MainPage renders correctly', async () => {
  render(
    <Provider store={store()}>
      <MainPage
        searchTerm="test"
        products={mockProducts}
        page={1}
        perPage={mockProducts.length}
      />
    </Provider>
  );

  const cards = await screen.findAllByTestId('card');

  expect(cards).toHaveLength(mockProducts.length);

  const inputElement = screen.getByRole('textbox') as HTMLInputElement;

  expect(inputElement.value).toBe('test');
});
