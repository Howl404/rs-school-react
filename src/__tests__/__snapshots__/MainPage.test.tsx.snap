// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders main page with all components and 2 products 1`] = `
<DocumentFragment>
  <button
    class="_errorButton_db53ca"
    data-testid="error-button"
  >
    Throw Error
  </button>
  <div
    class="_searchContainer_6ac984"
  >
    <input
      class="_searchInput_6ac984"
      type="text"
      value=""
    />
    <button
      class="_submitButton_6ac984"
    >
      Search
    </button>
  </div>
  <div
    class=""
  >
    <div
      class="_container_4eccd3"
      data-testid="card-container"
    >
      <div
        class="_card_c39673"
        data-testid="card"
      >
        <h3
          class="_cardHeading_c39673"
        >
          Product 1
        </h3>
        <img
          alt="Product 1 image"
          class="_cardImage_c39673"
          src=""
        />
        <p
          class="_cardText_c39673"
        />
      </div>
      <div
        class="_card_c39673"
        data-testid="card"
      >
        <h3
          class="_cardHeading_c39673"
        >
          Product 2
        </h3>
        <img
          alt="Product 2 image"
          class="_cardImage_c39673"
          src=""
        />
        <p
          class="_cardText_c39673"
        />
      </div>
    </div>
    <div
      class="_paginationContainer_5bac6f"
    >
      <button
        disabled=""
      >
        ᐸ
      </button>
      <p>
        1
      </p>
      <button>
        ᐳ
      </button>
      <select>
        <option
          value="5"
        >
          5 items per page
        </option>
        <option
          value="10"
        >
          10 items per page
        </option>
        <option
          value="15"
        >
          15 items per page
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
